src/main/java/com/google/starfish/servlets/SearchServlet.java

```diff
@@ -1,122 +1,54 @@
 
 import java.io.IOException;  
-import java.sql.Connection;  
 import java.sql.SQLException;  
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
 import javax.servlet.ServletException;  
 import javax.servlet.annotation.WebServlet;  
 import javax.servlet.http.HttpServlet;  
 import javax.servlet.http.HttpServletRequest;  
 import javax.servlet.http.HttpServletResponse;  
 import javax.sql.DataSource;
-import java.util.Date;
-import com.google.starfish.services.NoteService;
 import com.google.starfish.services.FavoriteNoteService;
-import com.google.starfish.models.Note;
-import com.google.gson.Gson;
-import java.util.ArrayList;
-import java.util.List;
+import com.google.starfish.services.FavoriteNoteService.Recency;
 
 /** Servlet that returns search results for notes based on school and course */
 @WebServlet("/search")  
 public class SearchServlet extends HttpServlet {  
 
-  private NoteService noteService = new NoteService();
   private FavoriteNoteService favoriteNoteService = new FavoriteNoteService();
 
   @Override
   public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
     DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");  
 
-    String reqSchool = trimAndLowerCaseString(req.getParameter("school"));
-    String reqCourse = trimAndLowerCaseString(req.getParameter("course"));
-
-    try (Connection conn = pool.getConnection()) {
-      try {
-        conn.setAutoCommit(false);
-        String stmt = getSearchQuery(reqSchool, reqCourse);
-        try (PreparedStatement getNotesStatement = conn.prepareStatement(stmt)) {
-          ResultSet rs = getNotesStatement.executeQuery();
-          conn.commit();
-
-          List<Note> notes = buildNotesFromQueryResult(pool, rs);
-          String json = convertListToJson(notes);
-          res.setContentType("application/json");
-          res.getWriter().println(json);
-        } 
-      } catch (SQLException ex) {
-          if (conn != null) {
-            try {
-              System.err.print("Transaction is being rolled back.");
-              conn.rollback();
-            } catch (SQLException excep) {
-              System.err.print(excep);
-            }
-          }
-        }
-      } catch (SQLException ex) {
-        System.err.print(ex);
+    String reqSchool = Utils.trimAndLowerCaseString(req.getParameter("school"));
+    String reqCourse = Utils.trimAndLowerCaseString(req.getParameter("course"));
+    String timespan = Utils.trimAndLowerCaseString(req.getParameter("timespan"));
+    Recency recency = Utils.findRecencyByString(timespan);
+    // Default recency is all-time
+    if (recency == null) recency = Recency.ALL_TIME;
+    try {
+      Object[][] trendingNotes = null;
+      switch(recency) {
+        case TODAY:
+          trendingNotes = favoriteNoteService.getTrendingNotesBySchoolOrCourse(pool, FavoriteNoteService.Recency.TODAY, reqSchool, reqCourse);
```

Comment:
same comment here as above file (but admittedly i'm not super familiar with java so might be wrong)

Reply:
Nope you're right, good catch!